sdk/couchbase/src/options/query_options.rs

```diff
@@ -1,10 +1,152 @@
+    pub client_context_id: Option<String>,
+    pub consistent_with: Option<MutationState>,
+    pub flex_index: Option<bool>,
+    pub max_parallelism: Option<u32>,
+    pub metrics: Option<bool>,
+    pub named_arguments: Option<HashMap<String, Value>>,
+    pub pipeline_batch: Option<u32>,
+    pub pipeline_cap: Option<u32>,
+    pub positional_parameters: Option<Vec<Value>>,
+    pub preserve_expiry: Option<bool>,
+    pub profile: Option<ProfileMode>,
+    pub raw: Option<HashMap<String, Value>>,
+    pub read_only: Option<bool>,
     pub retry_strategy: Option<Arc<dyn crate::retry::RetryStrategy>>,
+    pub scan_cap: Option<u32>,
+    pub scan_consistency: Option<ScanConsistency>,
+    pub scan_wait: Option<Duration>,
+    pub server_timeout: Option<Duration>,
+    pub use_replica: Option<ReplicaLevel>,
+}
+
+impl TryFrom<QueryOptions> for queryoptions::QueryOptions {
+    type Error = error::Error;
+
+    fn try_from(opts: QueryOptions) -> Result<queryoptions::QueryOptions, Self::Error> {
+        let (mutation_state, scan_consistency) = if let Some(mutation_state) = opts.consistent_with
+        {
+            (
+                Some(mutation_state.into()),
+                Some(queryx::query_options::ScanConsistency::AtPlus),
+            )
+        } else {
+            (None, opts.scan_consistency.map(|sc| sc.into()))
+        };
+
+        let named_args = if let Some(named_args) = opts.named_arguments {
+            let mut collected = HashMap::default();
+            for (k, v) in named_args {
+                collected.insert(k, serde_json::to_vec(&v)?);
+            }
+            Some(collected)
+        } else {
+            None
+        };
+        let raw = if let Some(raw) = opts.raw {
+            let mut collected = HashMap::default();
+            for (k, v) in raw {
+                collected.insert(k, serde_json::to_vec(&v)?);
+            }
+            Some(collected)
+        } else {
+            None
+        };
+        let positional_params = if let Some(positional_params) = opts.positional_parameters {
+            let mut collected = vec![];
+            for v in positional_params {
+                collected.push(serde_json::to_vec(&v)?);
+            }
+            Some(collected)
+        } else {
+            None
+        };
+
+        let mut builder = queryoptions::QueryOptions::builder()
+            .args(positional_params)
+            .client_context_id(opts.client_context_id)
+            .max_parallelism(opts.max_parallelism)
+            .metrics(opts.metrics)
+            .pipeline_batch(opts.pipeline_batch)
+            .pipeline_cap(opts.pipeline_cap)
+            .preserve_expiry(opts.preserve_expiry)
+            .profile(opts.profile.map(|p| p.into()))
+            .read_only(opts.read_only)
+            .scan_cap(opts.scan_cap)
+            .scan_consistency(scan_consistency)
+            .scan_wait(opts.scan_wait)
+            .sparse_scan_vectors(mutation_state)
```

Comment:
Is .sparse_scan_vectors supposed to be called with mutation_state as the input?

Reply:
Yup